### PR TITLE
Add majority voting ensemble in component nn classifier

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by johngian** [comment]
@marco-c r?

File: bugbug/models/component_nn.py [hunk 1]
```diff
@@ -178,7 +205,14 @@ def __init__(self, *args, **kwargs):
...
         }
-        self.clf = ComponentNNClassifier(**kwargs)
+
+        estimators = []
+        for i, param in enumerate(self.params):
+            kwargs['params'] = param
+            estimator = ComponentNNClassifier(**kwargs)
+            estimators.append(('model_{}'.format(i), estimator))
+
+        self.clf = VotingClassifier(estimators=estimators, voting='soft', weights=[1, 1, 1])
```
Comment (marco-c) [inline bugbug/models/component_nn.py]: If the VotingClassifier is doing the label encoding for us, then maybe we have to remove the encoding from KerasClassifier, or make it optional.

(I wonder why some classifiers do the encoding by themselves and some don't, it's painful!)

File: bugbug/models/component_nn.py [hunk 2]
```diff
@@ -30,59 +31,76 @@
...
         self.long_desc_vocab_size = kwargs.pop('long_desc_vocab_size')
         self.long_desc_emb_sz = kwargs.pop('long_desc_emb_sz')
+        self.params = kwargs.pop('params')
+        return self
 
     def model_creator(self, X, y):
         short_desc_inp = Input(shape=(self.short_desc_maxlen,), name='title_sequence')
         short_desc_emb = Embedding(self.short_desc_vocab_size, self.short_desc_emb_sz)(short_desc_inp)
-        short_desc_emb = SpatialDropout1D(0.2)(short_desc_emb)
+        short_desc_emb = SpatialDropout1D(self.params[0])(short_desc_emb)
```
Comment (marco-c) [inline bugbug/models/component_nn.py]: Could we make these params like the others, where you have names?

File: bugbug/model.py [hunk 3]
```diff
@@ -29,6 +29,14 @@ def __init__(self, lemmatization=False):
 
         self.calculate_importance = True
 
+    @property
+    def le(self):
```
Comment (marco-c) [inline bugbug/model.py]: I hate this, but I guess it's a necessary evil for now until we make the label encoding part better...

Comment (johngian) [reply bugbug/model.py]: We can always move it to the model definition and define it per model.

Comment (marco-c) [reply bugbug/model.py]: I mean I don't like the fact that we have to access the internal properties :(
I wish the classifiers didn't encode the labels and the responsibility of encoding were of the users of the classifiers.

**Comment by johngian** [comment]
@marco-c PR updated

File: bugbug/nn.py [hunk 4]
```diff
@@ -36,8 +35,7 @@ def __init__(self, epochs, batch_size):
     def fit(self, X, y):
         X_dict = numpy_to_dict(X)
 
-        self._le = LabelEncoder().fit(y)
```
Comment (marco-c) [inline bugbug/nn.py]: Maybe we have to remove the usage of _le in `predict` too?

**Comment by johngian** [comment]
@marco-c PR updated

**Pull request merged**